### ISSUE TITLE
Checkstyle-Plugin: SuppressionCommentFilter is not allowed as a child in Checker

### ISSUE THREAD

**Issue by SteVeRiGe** [checkstyle/eclipse-cs#134]
Hi,
I saw that this issue was reported already some time back, unfortunately the comments didn't help me solve my problem.
I installed the checkstyle plugin via Eclipse Marketplace (Version 8.12.0) in my Eclipse (Version: 2018-12 (4.10.0)). When running a checkstyle check, eclipse logs the following error in Log-File:
```
!ENTRY net.sf.eclipsecs.core 4 0 2019-02-15 17:03:01.793
!MESSAGE Checkstyle-Plugin: SuppressionCommentFilter is not allowed as a child in Checker
!STACK 0
com.puppycrawl.tools.checkstyle.api.CheckstyleException: SuppressionCommentFilter is not allowed as a child in Checker
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:480)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createCheckerInternal(CheckerFactory.java:299)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createChecker(CheckerFactory.java:133)
	at net.sf.eclipsecs.core.builder.Auditor.runAudit(Auditor.java:141)
	at net.sf.eclipsecs.core.builder.CheckstyleBuilder.handleBuildSelection(CheckstyleBuilder.java:306)
	at net.sf.eclipsecs.core.jobs.RunCheckstyleOnFilesJob.runInWorkspace(RunCheckstyleOnFilesJob.java:119)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:42)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

I did configure the plugin to use the "built-in" "google_checks.xml" and I also created my own checkstyle.xml containing nothing else than:

```
<module name="Checker">
	<property name="charset" value="UTF-8" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java, properties, xml" />
</module>
```
but unfortunately the error occured with both configurations. Running checkstyle using Maven-Plugin works perfectly fine, it's only the plugin which makes trouble for me. Any ideas how come around this?

**Comment by romani** [comment]
Error is valid, because Check's parent should be https://checkstyle.org/config_filters.html#SuppressionCommentFilter_Parent_Module

Parent was changed at https://checkstyle.org/releasenotes.html#Release_8.1

Please recheck versions of checkstyle library that you use in maven plugin.

You need to sync version to make behavior the same, you need to use https://checkstyle.org/config_filters.html#SuppressWithPlainTextCommentFilter if you really want to do suppression in non Java files.

**Comment by SteVeRiGe** [comment]
Hi Romani,
thanks for your response.
I'm a little bit confused. I do not use the SuppressionCommentFilter at all in my checkstyle.xml, but still get an error which tells me SuppressionCommentFilter is not allowed as a child in checker. I removed everything from my checkstyle.xml to a minimum, but still get this error. Here my minimum checkstyle.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	<property name="severity" value="error" />
</module>
```

I also checked the maven Plugin version. This is my maven config:
```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-checkstyle-plugin</artifactId>
				<version>3.0.0</version>
				<dependencies>
					<dependency>
						<groupId>com.puppycrawl.tools</groupId>
						<artifactId>checkstyle</artifactId>
						<version>8.12</version>
					</dependency>
				</dependencies>
				<configuration>
					<outputFileFormat>xml</outputFileFormat>
					<consoleOutput>true</consoleOutput>
					<includeTestSourceDirectory>true</includeTestSourceDirectory>
					<enableRSS>false</enableRSS>
					<linkXRef>false</linkXRef>
					<configLocation>checkstyle.xml</configLocation>
					<failOnViolation>true</failOnViolation>
					<violationSeverity>error</violationSeverity>
				</configuration>
				<executions>
					<execution>
						<id>validate</id>
						<phase>validate</phase>
						<goals>
							<goal>check</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
```
Did I miss anything?

**Comment by melloware** [comment]
I am getting a similar error I was not getting before with the same config file.

{code}
MESSAGE Checkstyle-Plugin: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
!STACK 0
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:457)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createCheckerInternal(CheckerFactory.java:290)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createChecker(CheckerFactory.java:128)
	at net.sf.eclipsecs.core.builder.Auditor.runAudit(Auditor.java:141)
	at net.sf.eclipsecs.core.builder.CheckstyleBuilder.handleBuildSelection(CheckstyleBuilder.java:301)
	at net.sf.eclipsecs.core.jobs.RunCheckstyleOnFilesJob.runInWorkspace(RunCheckstyleOnFilesJob.java:119)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:42)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
	at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:155)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:452)
	... 8 more
{code}

**Comment by romani** [comment]
We need community help to reproduce issue, need some sort of steps to reproduce

**Comment by lkoe** [comment]
Could you please attach the .checkstyle file from one of the affected projects?

I'd expect this error only to occur when using an older configuration with current Checkstyle, where the module has been moved to the Checker parent.

**Comment by melloware** [comment]
Attached is my checkstyle.xml of my open souce project.



[checkstyle.zip](https://github.com/checkstyle/eclipse-cs/files/2999593/checkstyle.zip)

**Comment by lkoe** [comment]
@melloware this checkstyle configuration works alright for me

Please check the plugin version you are using (I've just tested with the latest 8.18.0)

**Comment by melloware** [comment]
It works for me too in the Maven Plugin but i was getting the error above in Eclipse using this same configuration.  I had wiped my Eclipse and started over and now I am not seeing this error.

**Comment by romani** [comment]
I am closing this issue, if  somebody come up with steps to reproduce, please open new issue.
